Mage.Sets/src/mage/cards/c/CemeteryProtector.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    private CemeteryProtector(final CemeteryProtector card) {
+        super(card);
+    }
+
+    @Override
+    public CemeteryProtector copy() {
+        return new CemeteryProtector(this);
+    }
+}
+
+class CemeteryProtectorEffect extends OneShotEffect {
+
+    public CemeteryProtectorEffect() {
+        super(Outcome.Exile);
+        staticText = "exile a card from a graveyard";
+    }
+
+    private CemeteryProtectorEffect(final CemeteryProtectorEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public CemeteryProtectorEffect copy() {
+        return new CemeteryProtectorEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            TargetCardInGraveyard target = new TargetCardInGraveyard();
+            target.setNotTarget(true);
+            controller.choose(outcome, target, source.getSourceId(), game);
+            Card card = game.getCard(target.getFirstTarget());
+            if (card != null) {
+                UUID exileId = CardUtil.getExileZoneId(game, source.getSourceId(), source.getSourceObjectZoneChangeCounter());
+                MageObject sourceObject = source.getSourceObject(game);
+                String exileName = sourceObject == null ? null : sourceObject.getIdName();
+                return controller.moveCardsToExile(card, source, game, true, exileId, exileName);
+            }
+        }
+        return false;
+    }
+}
+
+class CemeteryProtectorTriggeredAbility extends TriggeredAbilityImpl {
+
+    private UUID idToCheck = null;
+    private GameEvent.EventType eventToCheck = null;
+    private UUID exileZoneId = null;
+
+    public CemeteryProtectorTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new CreateTokenEffect(new HumanToken()));
+    }
+
+    private CemeteryProtectorTriggeredAbility(final CemeteryProtectorTriggeredAbility ability) {
+        super(ability);
+        idToCheck = ability.idToCheck;
+        eventToCheck = ability.eventToCheck;
+        exileZoneId = ability.exileZoneId;
+    }
+
+    @Override
+    public CemeteryProtectorTriggeredAbility copy() {
+        return new CemeteryProtectorTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.LAND_PLAYED || event.getType() == GameEvent.EventType.SPELL_CAST;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (event.getPlayerId().equals(controllerId)) {
+            idToCheck = event.getTargetId();
+            eventToCheck = event.getType();
+            // TODO: sourceObjectZoneChangeCounter is incorrect here (always reporting 0).  Workaround is to get ZCC from GameState.
+            // May need research as to why this variable isn't reporting correctly.
```

Comment:
Cause ability's zcc is static and filled on ability activation (e.g. after check and raise for triggers).